pkg/cmd/get_clustercompliancereport.go

```diff
@@ -0,0 +1,118 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/starboard/pkg/compliance"
+	"github.com/aquasecurity/starboard/pkg/starboard"
+	"github.com/spf13/cobra"
+	"io"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+func NewGetClusterComplianceReportsCmd(executable string, cf *genericclioptions.ConfigFlags, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "clustercompliancereports (NAME)",
+		Aliases: []string{"clustercompliance"},
+		Short:   "Get cluster compliance reports",
+		Long:    `Get cluster compliance report for pre-define spec`,
+		Example: fmt.Sprintf(`
+# Get cluster compliance report for specifc spec in JSON output format
+  %[1]s get clustercompliancereports nsa -o json
+
+# Get compliance detail report for control checks failure in JSON output format
+  %[1]s get clustercompliancereports nsa -o json --detail`, executable),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			//init client dependencies
+			logger := ctrl.Log.WithName("reconciler").WithName("clustercompliancereport")
+			ctx := context.Background()
+			scheme := starboard.NewScheme()
+			kubeConfig, err := cf.ToRESTConfig()
+			if err != nil {
+				return fmt.Errorf("failed to create kubeConfig: %w", err)
+			}
+			// create k8s client
+			kubeClient, err := client.New(kubeConfig, client.Options{Scheme: scheme})
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %w", err)
+			}
+			namespaceName, err := ComplianceNameFromArgs(args)
+			if err != nil {
+				return err
+			}
+			// check report if report with spec exist
+			var report v1alpha1.ClusterComplianceReport
+			err = GetComplianceReport(ctx, kubeClient, namespaceName, out, &report)
+			if err != nil {
+				return err
+			}
+			// generate compliance and compliance failure detail reports
+			complianceMgr := compliance.NewMgr(kubeClient, logger)
+			err = complianceMgr.GenerateComplianceReport(ctx, report.Spec)
+			if err != nil {
+				return fmt.Errorf("failed to generate report: %w", err)
+			}
+
+			format := cmd.Flag("output").Value.String()
+			printer, err := genericclioptions.NewPrintFlags("").
+				WithTypeSetter(scheme).
+				WithDefaultOutput(format).
+				ToPrinter()
+			if err != nil {
+				return fmt.Errorf("faild to create printer: %w", err)
+			}
+			// check report detail flag has been set
+			detailString := cmd.Flag("detail").Value.String()
+			detail, err := strconv.ParseBool(detailString)
+			if err != nil {
```

Comment:
Error while parsing flag should be propagated to the caller and eventually printed to stderr. Here we fallback to false value